catalogue_pipeline/transformer/src/test/scala/uk/ac/wellcome/transformer/transformers/sierra/SierraContributorsTest.scala

```diff
@@ -0,0 +1,662 @@
+    }
+
+    it("gets an identifier from subfield $$0") {
+      val name = "Ivan the ivy"
+      val lcshCode = "lcsh7101607"
+
+      val varFields = List(
+        VarField(
+          fieldTag = "p",
+          marcTag = "100",
+          indicator1 = "",
+          indicator2 = "",
+          subfields = List(
+            MarcSubfield(tag = "a", content = name),
+            MarcSubfield(tag = "0", content = lcshCode)
+          )
+        )
+      )
+
+      val sourceIdentifier = SourceIdentifier(
+        identifierScheme = IdentifierSchemes.libraryOfCongressNames,
+        ontologyType = "Person",
+        value = lcshCode
+      )
+
+      val expectedContributors = List(
+        Contributor(
+          agent = Identifiable(
+            Person(label = name),
+            sourceIdentifier = sourceIdentifier,
+            identifiers = List(sourceIdentifier)
+          ))
+      )
+
+      transformAndCheckContributors(
+        varFields = varFields,
+        expectedContributors = expectedContributors)
+    }
+
+    it("gets an identifier with inconsistent spacing from subfield $$0") {
+      val name = "Wanda the watercress"
+      val lcshCodeCanonical = "lcsh2055034"
+      val lcshCode1 = "lcsh 2055034"
+      val lcshCode2 = "  lcsh2055034 "
+      val lcshCode3 = " lc sh 2055034"
+
+      val varFields = List(
+        VarField(
+          fieldTag = "p",
+          marcTag = "100",
+          indicator1 = "",
+          indicator2 = "",
+          subfields = List(
+            MarcSubfield(tag = "a", content = name),
+            MarcSubfield(tag = "0", content = lcshCode1),
+            MarcSubfield(tag = "0", content = lcshCode2),
+            MarcSubfield(tag = "0", content = lcshCode3)
+          )
+        )
+      )
+
+      val sourceIdentifier = SourceIdentifier(
+        identifierScheme = IdentifierSchemes.libraryOfCongressNames,
+        ontologyType = "Person",
+        value = lcshCodeCanonical
+      )
+
+      val expectedContributors = List(
+        Contributor(
+          agent = Identifiable(
+            Person(label = name),
+            sourceIdentifier = sourceIdentifier,
+            identifiers = List(sourceIdentifier)
+          ))
+      )
+
+      transformAndCheckContributors(
+        varFields = varFields,
+        expectedContributors = expectedContributors)
+    }
+
+    it(
+      "fails the transform if there are multiple distinct identifiers in subfield $$0") {
+      val varFields = List(
+        VarField(
+          fieldTag = "p",
+          marcTag = "100",
+          indicator1 = "",
+          indicator2 = "",
+          subfields = List(
+            MarcSubfield(tag = "a", content = "Darren the Dill"),
+            MarcSubfield(tag = "0", content = "lcsh9069541"),
+            MarcSubfield(tag = "0", content = "lcsh3384149")
+          )
+        )
+      )
+
+      assertTransformFails(varFields = varFields)
+    }
+  }
+
+  describe("Organisation") {
```

Comment:
This test class is a bit huge. Could we move the person tests into a `SierraContributorsPersonsTest` (or something like that) and do the same for organisations?

Reply:
For the sake of getting this merged, I’ll leave it for now – we can break it up if/when we’re next in this part of the codebase.